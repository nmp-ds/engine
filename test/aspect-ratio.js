import { setup } from './_helpers.js';
import { expect, test } from 'vitest';

setup();

test('it generates backported aspect-ratios', async ({ uno }) => {
  const classes = ['aspect-16/9', 'aspect-1/1', 'aspect-4/3', 'aspect-2/1'];
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('aspect-ratio, -video and -square', async ({ uno }) => {
  const classes = ['aspect-ratio', 'aspect-video', 'aspect-square'];
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});