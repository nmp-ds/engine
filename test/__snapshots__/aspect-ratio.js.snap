// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`aspect-ratio, -video and -square 1`] = `
"/* layer: default */
.aspect-square{position:relative;padding-bottom:100%;}.aspect-square>*{position:absolute;height:100%;width:100%;top:0;right:0;bottom:0;left:0;}
.aspect-video{position:relative;padding-bottom:56.25%;}.aspect-video>*{position:absolute;height:100%;width:100%;top:0;right:0;bottom:0;left:0;}
.aspect-ratio{aspect-ratio:auto;}"
`;

exports[`it generates backported aspect-ratios 1`] = `
"/* layer: default */
.aspect-1\\\\/1{position:relative;padding-bottom:100%;}.aspect-1\\\\/1>*{position:absolute;height:100%;width:100%;top:0;right:0;bottom:0;left:0;}
.aspect-16\\\\/9{position:relative;padding-bottom:56.25%;}.aspect-16\\\\/9>*{position:absolute;height:100%;width:100%;top:0;right:0;bottom:0;left:0;}
.aspect-2\\\\/1{position:relative;padding-bottom:50%;}.aspect-2\\\\/1>*{position:absolute;height:100%;width:100%;top:0;right:0;bottom:0;left:0;}
.aspect-4\\\\/3{position:relative;padding-bottom:75%;}.aspect-4\\\\/3>*{position:absolute;height:100%;width:100%;top:0;right:0;bottom:0;left:0;}"
`;
